scalding-core/src/main/scala/com/twitter/scalding/Job.scala

```diff
@@ -521,11 +521,11 @@ trait CounterVerification extends Job {
   }
 }
 
-private[scalding] object FlowStepStrategies {
+private[scalding] case class FlowStepStrategies[A]() extends Semigroup[FlowStepStrategy[A]] {
```

Comment:
Making an analogy to Haskell, it's equivalent to the semigroup over this type:

```haskell
newtype FlowStepStrategy a =
    FlowStepStrategy (Flow a -> JList (FlowStep a) -> FlowStep a -> IO ())

instance Monoid (FlowStepStrategy a) where
    mempty = FlowStepStrategy (\_ _ _ -> return ())

    mappend (FlowStepStrategy f1) (FlowStepStrategy f2) = FlowStepStrategy f
      where
        f x y z = do
            f1 x y z
            f2 x y z
```

... and that's guaranteed to satisfy the `Monoid` laws in both Haskell and Scala.

The most common source of this thing failing in Scala specifically is if side effects are triggered prematurely by evaluating things in the wrong order but that does not appear to be the case for this code.